packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -15,9 +19,10 @@ import { useRef, useCallback, useLayoutEffect } from '@wordpress/element';
  * callback will be called with `null` and the new ref callback will be called
  * with the same node.
  *
- * @param {Array<RefObject|RefCallback>} refs The refs to be merged.
+ * @template {import('react').Ref<any>} T
+ * @param {Array<T>} refs The refs to be merged.
  *
- * @return {RefCallback} The merged ref callback.
+ * @return {import('react').RefCallback<TypeFromRef<T>>} The merged ref callback.
```

Comment:
Also I tried to look up `RefCallback` but couldn't find anything. Any pointers?

Reply:
`RefCallback` is the style of `ref` prop that goes like: `<div ref={ node => { /* do something with node, probably assign it to this.something */ } } />`. The type is literally just a function type but with a specific name used for `ref` props (as is the intended usage of the return type of this hook).